src/dune_rules/modules.ml

```diff
@@ -1264,3 +1264,13 @@ let canonical_path t (group : Group.t) m =
   | Impl { impl = { modules = Wrapped w; _ }; _ } | Wrapped w -> w.group.name :: path
   | _ -> Module.path m
 ;;
+
+type enabled =
```

Comment:
The only reason why I added this type and fun here is because both `Ml_sources` and `Artifacts_obj` depend on it. `Artifacts_obj` didn't seem like the right place, and adding it to `Ml_sources` would create a cycle.

Reply:
We have a `Toggle.t` type that's the same as this. I'd say you can just use it